packages/components/src/components/TaskRunDetails/TaskRunDetails.js

```diff
@@ -95,6 +180,7 @@ const TaskRunDetails = ({ intl, onViewChange, pod, task, taskRun, view }) => {
   const tabs = [
     paramsTable && 'params',
     resultsTable && 'results',
+    showIO && 'resources',
```

Comment:
We should only show the Resources tab if the `showIO` flag is enabled AND we actually have resources to display.